### PR TITLE
Test 6 p1

### PR DESCRIPTION
**Pull request by matix522**

### CODE REVIEW

**Comment by tomekzaw** [comment]
Can we close the rest of your PRs and just merge this one?

File: tests/project/test_task1.py [hunk 1]
```diff
@@ -19,5 +23,144 @@ def test_P1():
     pass
 
 
+def check_production(G: Graph, e: Node):
```
Comment (tomekzaw) [inline tests/project/test_task1.py]: The name of this function is kinda unclear to me.

File: tests/project/test_task1.py [hunk 2]
```diff
@@ -19,5 +23,144 @@ def test_P1():
...
+            if neighbour.label == "E":
+                assert neighbour in Es
+                assert bool(neighbour.x == E.x) != bool(neighbour.y == E.y)
+            else:
+                assert neighbour.label == "I"
+                E_not_neighbours = [E1 for E1 in Es if E1 not in G.neighbors(E) and E1 != E]
+                assert len(E_not_neighbours) == 1
+                E_not_neighbour = E_not_neighbours[0]
+                assert (E_not_neighbour.x + E.x) / 2 == neighbour.x
+                assert (E_not_neighbour.y + E.y) / 2 == neighbour.y
```
Comment (tomekzaw) [inline tests/project/test_task1.py]: You can simply use `is_node_between` utility function like so:
```suggestion
                assert is_node_between(E_not_neighbour, neighbour, E)
```

File: tests/project/test_task1.py [hunk 3]
```diff
@@ -19,5 +23,144 @@ def test_P1():
...
+
+
+def test_P1_vertex_removed():
+    """Empty graph"""
+    G = Graph()
+
+    assert G.number_of_nodes == 0
+
+    with pytest.raises(CannotApplyProductionError):
+        P1(G)
```
Comment (tomekzaw) [inline tests/project/test_task1.py]: You can use `assert_production_cannot_be_applied` test utility function by @pk1890 here.

**Review by tomekzaw — CHANGES_REQUESTED** [review]
Looks good to me. A few minor code style issues but in general it's OK.

File: tests/project/test_task1.py (hunk 1 above)
Comment (matix522) [reply tests/project/test_task1.py]: Thats on purpose

File: tests/project/test_task1.py (hunk 2 above)
Comment (matix522) [reply tests/project/test_task1.py]: eehhh..

File: tests/project/test_task1.py (hunk 3 above)
Comment (matix522) [reply tests/project/test_task1.py]: done

**Review by tomekzaw — APPROVED** [review]
Looks good to me! 🚀

**Pull request merged**